esprima.js

```diff
@@ -187,6 +188,7 @@
         IllegalContinue: 'Illegal continue statement',
         IllegalBreak: 'Illegal break statement',
         IllegalReturn: 'Illegal return statement',
+        IllegalYield: 'Illegal yield expression',
```

Comment:
For sake of understanding -- what systems might break if we deviated a bit from v8 error messages?

I ask because this seems like a strictly more useful error message for the user + it seems like the idea of guaranteeing that errors match is a bit of a moving target (i.e. don't these things change over time in V8?)